src/socket.rs

```diff
@@ -138,30 +144,28 @@ impl Socket {
     //                 the next invocation of the ready handler.
     //   - Err(error): there was an error while writing to the socket.
     pub fn flush(&mut self) -> Result<bool, SocketError> {
-        if let Some(mut data) = self.write_queue.pop_front() {
+        while let Some(mut data) = self.write_queue.pop_front() {
```

Comment:
The previous `if` is what is ideal here. In case the messages accumulate and then later when socket is writable we still dont want to write all messages in one go. Infact we dont even want to write single msg in one go. Only as much as socket.write(..) writes in one go. So this should be interleaved. Pop front .. write .. if all went then fine else push the remaining part of msg push front .. check if queue is empty .. if not ok(false) else Ok(true). That way we keep everything interleaved.